lib/topology_openswitch/vtysh.py

```diff
@@ -170,8 +171,27 @@ def _determine_set_prompt(self, connection=None):
         # always return the produced output, even if an EOF exception
         # follows after it. This is done to handle the segmentation fault
         # errors.
-        spawn.sendline('stdbuf -oL vtysh')
-        spawn.expect(VTYSH_STANDARD_PROMPT)
+        for i in range(10):
+                try:
+                    spawn.sendline('stdbuf -oL vtysh')
+                    spawn.expect(VTYSH_STANDARD_PROMPT, timeout=3)
+                    break
+                except TIMEOUT:
+                    continue
+                except Exception as error:
+                    warning(
```

Comment:
Raise the error here instead of a warning. Imagine that we hit this line of code 10 times. The next thing that happens is that the user is informed of the ``before`` attribute of the spawn object which is unrelated to the actual error that is the important thing to report.